libcalico-go/lib/backend/k8s/resources/node.go

```diff
@@ -230,7 +232,11 @@ func K8sNodeToCalico(k8sNode *kapiv1.Node, usePodCIDR bool) (*model.KVPair, erro
 	annotations := k8sNode.ObjectMeta.Annotations
 	bgpSpec.IPv4Address = annotations[nodeBgpIpv4AddrAnnotation]
```

Comment:
I'd personally be in favor of using the logic we just wrote to also validate the other IP annotations - v4 / v6 addresses. They have the same behavior. 

There are a few others below as well - Wireguard IPs, tunnel IPs